docs/cosmos-based/parser/migrations/v3.0.0.md

```diff
@@ -0,0 +1,164 @@
+or leave it empty if no custom address parser is needed. 
+```go
+// If no custom address parser
+var CustomAccountParser = []string{}
+```
+```go
+// Take desmos chain as example:
+var CustomAccountParser = []string{
+  "sender", "receiver", "user", "counterparty", "blocker", "blocked",
+}
+```
+2. Checkout inside the **/bdjuno** directory and run:
+```shell
+$ make install
+$ bdjuno migrate v3
+```
+Two additional fields inside `.bdjuno/config.yaml` file will be created under the database section. 
+Don't panic if you see the error message pops up: `partition batch size is set to 0, skipping migration`, 
+because these 2 fields need to be modified before starting the migration, for example:
+```yaml
+database:
+    # How many blocks for each partition. 
+    # In this example partition is created per 100,000 blocks.
+    partition_size: 100000
+    
+    # How many rows of transactions are migrated per batch 
+    partition_batch: 1000 
+```
+Read [Database Config Reference](./../config/config.md#database) for more information.
+
+### Start the migration
+**Make sure bdjuno is stopped before the migration.**  
+We recommend setting up the migration process as a system service and being patient 
+as the migration might take some time, depending on the database size.
+
+```shell
+$ sudo tee /etc/systemd/system/bdjuno-migrate-v3.service > /dev/null <<EOF
+[Unit]
+Description=BDJuno Table Partition Migration v3
+After=network-online.target
+
+[Service]
+User=$USER
+ExecStart=$GOPATH/bin/bdjuno migrate v3
+LimitNOFILE=4096
+
+[Install]
+WantedBy=multi-user.target
+EOF
+```
+
+Then you need to enable and start the service:
+```shell 
+$ sudo systemctl enable bdjuno-migrate-v3.service
+$ sudo systemctl start bdjuno-migrate-v3.service
+```
+
+BDJuno will now start migration which will:  
+- rename `transaction` & `message` tables to `transaction_old` & `message_old`  
+- rename related transaction and message indexes; 
+- create new `transaction` & `message` tables with partition;
+- create indexes for new transaction and message table  
+- migrate rows with data by batch from old to new tables;
+- drop old function `messages_by_address` and create a new one;
+
+While the migration is ongoing, you can already restart bdjuno to start 
+syncing the latest data and it will be stored inside new partitioned tables:
+```shell
+$ bdjuno start
+```
+You will also need to re-apply hasura metadata to match the latest table setup:
+```shell
+$ cd /path/to/bdjuno/hasura
+$ hasura metadata apply --endpoint <your-endpoint> --admin-secret <hasura_password>
+```
+
+
+### Migration completed
+Once all transactions and messages have been migrated to new partitioned tables the migration is completed. For the safety reasons 
+it does __not__ drop old tables `transaction_old` and `message_old`. 
+You might want to drop them after the data consistency is verified and they are no longer needed.   
+```sql
+DROP TABLE transaction_old;
+DROP TABLE message_old;
+```
+
```

Comment:
I think that here we should also tell the user to update the BDJuno service content to replace the `parse` command with `start` instead